cyral/resource_cyral_repository_local_account.go

```diff
@@ -121,31 +121,86 @@ func (resource *HashicorpVaultResource) ReadFromSchema(d *schema.ResourceData) {
 	}
 }
 
-type EnviromentVariableResource struct {
+type EnvironmentVariableResource struct {
 	DatabaseName string `json:"databaseName"`
 	RepoAccount  string `json:"repoAccount"`
 	VariableName string `json:"variableName"`
 }
 
-func (resource EnviromentVariableResource) WriteToSchema(d *schema.ResourceData) {
-	d.Set("enviroment_variable", []interface{}{
+func (resource EnvironmentVariableResource) WriteToSchema(d *schema.ResourceData) {
+	_, useDeprecatedArgument := d.GetOk("enviroment_variable")
+
+	if useDeprecatedArgument { // Deprecated: should be removed in the next MAJOR release
+		d.Set("enviroment_variable", []interface{}{
+			map[string]interface{}{
+				"database_name": resource.DatabaseName,
+				"local_account": resource.RepoAccount,
+				"variable_name": resource.VariableName,
+			},
+		})
+	} else {
+		d.Set("environment_variable", []interface{}{
+			map[string]interface{}{
+				"database_name": resource.DatabaseName,
+				"local_account": resource.RepoAccount,
+				"variable_name": resource.VariableName,
+			},
+		})
+	}
```

Comment:
Had a hard time finding the actual difference and saw that it was the typo in `environment`. I think the best make this code easier to follow is to just have in the conditional the two strings with and without the typo as follows:
```suggestion
	argument := "environment_variable"
	if useDeprecatedArgument {
		// Deprecated: should be removed in the next MAJOR release
		argument = "enviroment_variable"
	}
	d.Set(argument, []interface{}{
		map[string]interface{}{
			"database_name": resource.DatabaseName,
			"local_account": resource.RepoAccount,
			"variable_name": resource.VariableName,
		},
	})
```

Reply:
That's way better, thanks for the suggestion!